### PR TITLE
[openwebnet] rename thermoregulation's (WHO=4) things

### LINKED ISSUE #10849: [openwebnet] rename thermoregulation's (WHO=4) things
Because it is more correct to identify an environment with the concept of `zone` rather than  `thermostat`(as a `zone` is composed of at least one `thermostat` and at least one `actuator`) `bus_thermostat` and `bus_temp_sensor` should be renamed to `bus_thermo_zone` and `bus_thermo_sensor`

### PR DESCRIPTION
**Pull request by aconte80**
Renamed things used in WHO=4 Thermoregulation.

*** BREAKING CHANGE ***
`bus_thermostat`  renamed to `bus_thermo_zone` 
`bus_temp_sensor` renamed to `bus_thermo_sensor`

Updated openwebnet4j to 0.5.3.

Fixes #10849

### CODE REVIEW

**Comment by aconte80** [comment]
@fwolter this PR is just a refactoring of naming conventions of things associated to Thermoregulation: we restored support for thermoregulation in milestone 5 but, after a lot of discussions even with the community, we decide to apply this renaming.
Because it's a breaking change against what users are used to do in OH2 it would be better if we came out with the new nomenclature in OH 3.1. Do you think it's feasible?

**Comment by mvalla** [comment]
yes @fwolter: if current (old) thermo Things naming goes to 3.1, then users will have to move to the new naming in 3.2, while if this PR is included in 3.1 they will directly use the final correct naming when they move their thermo Things to OH3.

Also: after you accept this PR, I can proceed with the refactoring to move everthing under the internal package (as you suggested), so everthing would be in place in 3,1 also from the package point of view.

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 1]
```diff
@@ -171,22 +188,22 @@ It's possible to enter a value manually or set `shutterRun=AUTO` (default) to ca
 
 BUS gateway and things configuration:
 
-```xtend
+```json
```
Comment (fwolter) [inline bundles/org.openhab.binding.openwebnet/README.md:191]: Please remove this at all. The type doesn't need to be specified any longer.

**Review by fwolter — CHANGES_REQUESTED** [review]
If I understand correctly, this is a breaking change from 3.0 to 3.1. So you could add an ALTERT entry to https://github.com/openhab/openhab-distro/blob/main/distributions/openhab/src/main/resources/bin/update.lst.

**Comment by mvalla** [comment]
> If I understand correctly, this is a breaking change from 3.0 to 3.1. So you could add an ALTERT entry to https://github.com/openhab/openhab-distro/blob/main/distributions/openhab/src/main/resources/bin/update.lst.
  
  Actually Thermo support would be introduced just now in the official openwebnet binding with 3.1: in 2.5.x or 3.0.x is not there yet. 
  The  "breaking change" @aconte80 mentions is only in comparison to the unofficial testing version of the openwebnet binding (which works on 2.5 core). 
  So in my opinion, if this PR get merged into 3.1, there is no need for an official breaking change alert: we can just put it clear in the community thread that "starting from 3.1 the Thermo part is supported, but Things have a new name from the old testing binding".

**Comment by fwolter** [comment]
OK, that's indeed no breaking change.

File: bundles/org.openhab.binding.openwebnet/README.md (hunk 1 above)
Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/README.md]: removed these type tags

**Review by fwolter — APPROVED** [review]
LGTM

**Pull request merged**